task/buildah-oci-ta/0.2/buildah-oci-ta.yaml

```diff
@@ -20,6 +20,11 @@ spec:
       description: Name of secret which contains subscription activation key
       type: string
       default: activation-key
+    - name: ADDITIONAL_LABELS
+      description: Additional key=value labels that should be applied to the
+        image
+      type: string
+      default: '[]'
```

Comment:
I tried for a while to make this of `type: array`, but I couldn't figure out how to get an array typed param into the task script as an env var. The only other array type param we have gets supplied as `args`, and I don't think we can do that with _two_ params.

Here, I'm treating it as a json string.

Reply:
I do think an array of `--label` args would be nicer. This should probably work:

```diff
diff --git a/task/buildah/0.2/buildah.yaml b/task/buildah/0.2/buildah.yaml
index 0c992bbc..b2fe87b0 100644
--- a/task/buildah/0.2/buildah.yaml
+++ b/task/buildah/0.2/buildah.yaml
@@ -177,10 +177,34 @@ spec:
     - name: COMMIT_SHA
       value: $(params.COMMIT_SHA)
     args:
+      - --build-args
       - $(params.BUILD_ARGS[*])
+      - --labels
+      - $(params.LABELS[*])
     script: |
       #!/bin/bash
       set -e
+
+      build_args=()
+      labels=()
+
+      while [ $# -gt 0 ]; do
+          case $1 in
+              --build-args)
+                  shift
+                  while [[ $# -gt 0 && $1 != --* ]]; do build_args+=("$1"); shift; done
+                  ;;
+              --labels)
+                  shift
+                  while [[ $# -gt 0 && $1 != --* ]]; do labels+=("$1"); shift; done
+                  ;;
+              *)
+                  echo "unexpected argument: $1" >&2
+                  exit 2
+                  ;;
+          esac
+      done
+
       ca_bundle=/mnt/trusted-ca/ca-bundle.crt
       if [ -f "$ca_bundle" ]; then
         echo "INFO: Using mounted CA bundle: $ca_bundle"
```

Reply:
OK. Update in place now at https://github.com/konflux-ci/build-definitions/commit/5da7b94f6ded2317d3fe1d9f755420900d067b80.